functions/supplier/geode-cq-supplier/src/test/java/org/springframework/cloud/fn/supplier/geode/cq/GeodeCqSupplierApplicationTests.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.geode.cache.Region;
+import org.apache.geode.pdx.PdxInstance;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import reactor.core.publisher.Flux;
+import reactor.test.StepVerifier;
+
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.cloud.fn.common.geode.JsonPdxFunctions;
+import org.springframework.cloud.fn.test.support.geode.GeodeContainer;
+import org.springframework.cloud.fn.test.support.geode.GeodeContainerIntializer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class GeodeCqSupplierApplicationTests {
+
+	private ApplicationContextRunner applicationContextRunner;
+
+	private GeodeContainer geode;
+
+	private ObjectMapper objectMapper = new ObjectMapper();
+
+	@BeforeAll
+	void setup() {
+		GeodeContainerIntializer initializer = new GeodeContainerIntializer(
+				geodeContainer -> {
+					geodeContainer.connectAndExecGfsh("create region --name=Stocks --type=REPLICATE");
+				});
+
+		applicationContextRunner = new ApplicationContextRunner()
+				.withUserConfiguration(GeodeCqSupplierTestApplication.class);
+
+		geode = initializer.geodeContainer();
+	}
+
+	@Test
+	void pdxReadSerialized() {
+		applicationContextRunner
+				.withPropertyValues(
+						"geode.region.regionName=Stocks",
+						"geode.client.pdx-read-serialized=true",
+						"geode.cq.supplier.query= Select * from /Stocks where symbol='XXX' and price > 140",
+						"geode.pool.hostAddresses=" + "localhost:" + geode.getLocatorPort())
+				.run(context -> {
+					Supplier<Flux<String>> geodeCqSupplier = context.getBean("geodeCqSupplier", Supplier.class);
+					// Using local region here
+					Region<String, PdxInstance> region = context.getBean(Region.class);
+					putStockEvent(region, new Stock("XXX", 140.00));
+					putStockEvent(region, new Stock("XXX", 140.20));
+					putStockEvent(region, new Stock("YYY", 110.00));
+					putStockEvent(region, new Stock("YYY", 110.01));
+					putStockEvent(region, new Stock("XXX", 139.80));
+
+					StepVerifier.create(geodeCqSupplier.get()).assertNext(val -> {
+						try {
+							assertThat(objectMapper.readValue(val, Stock.class)).isEqualTo(new Stock("XXX", 140.20));
+						}
+						catch (JsonProcessingException e) {
+							fail(e.getMessage());
+						}
+					}).thenCancel().verify(Duration.ofSeconds(10));
+				});
+	}
+
+	private void putStockEvent(Region<String, PdxInstance> region, Stock stock) throws JsonProcessingException {
+		String json = objectMapper.writeValueAsString(stock);
+		region.put(stock.getSymbol(), JsonPdxFunctions.jsonToPdx().apply(json));
+	}
+
+	@Data
+	@AllArgsConstructor
+	@NoArgsConstructor
+	static class Stock {
+		private String symbol;
+
+		private double price;
+	}
```

Comment:
Can we de-lombok?

Reply:
Why? So I can do more work?

Reply:
The consensus on the Spring team is we don't use Lombok in framework code, but ok for samples, tests. etc.  It's just a convenience here and test scope doesn't violate this.